utils/debug/Debug.js

```diff
@@ -0,0 +1,128 @@
+
+    // line graph for nb elements
+    const nbObjectsCanvas = document.createElement('canvas');
+    nbObjectsCanvas.heigth = '20rem';
+    nbObjectsCanvas.width = '40rem';
+    nbObjectsCanvas.id = 'nb-objects';
+    chartDiv.appendChild(nbObjectsCanvas);
+
+    const timestamp = Date.now();
+    const nbObjectsDataset = { label: 'Number of object in Scene', data: [{ x: 0, y: 0 }] };
+    const nbObjectsChartLabel = ['0s'];
+    const nbObjectsChart = new Chart('nb-objects', {
+        type: 'line',
+        data: {
+            labels: nbObjectsChartLabel,
+            datasets: [nbObjectsDataset],
+        },
+        options: {
+            animation: { duration: 10 },
+            scales: {
+                yAxes: [{
+                    display: true,
+                    ticks: {
+                        suggestedMin: 0, // minimum will be 0, unless there is a lower value.
+                    },
+                }],
+            },
+        },
+    });
+
+    function updateNbObjectsChart() {
+        function countElem(node) {
+            if (!node) {
+                return 0;
+            }
+            let count = 1; // this node
+            if (node.children) {
+                for (const child of node.children) {
+                    count += countElem(child);
+                }
+            }
+            return count;
+        }
+
+        const timeInS = Math.floor((Date.now() - timestamp) / 1000);
+
+        const limit = 50;
+        nbObjectsChartLabel.push(`${timeInS}s`);
+        if (nbObjectsChartLabel.length > limit) {
+            nbObjectsChartLabel.shift();
+        }
+        nbObjectsDataset.data.push({ x: timeInS, y: countElem(scene.gfxEngine.scene3D) });
+        if (nbObjectsDataset.data.length > limit) {
+            nbObjectsDataset.data.shift();
+        }
+        nbObjectsChart.update();
+    }
+
+    // hook that to scene.update
+    const oldUpdate = Object.getPrototypeOf(scene).update.bind(scene);
+    scene.update = function debugUpdate() {
+        oldUpdate();
+        updateNbObjectsChart();
+    };
+
+    // DEBUG CONTROLS
+    const gui = menuGlobe.gui.addFolder('Debug Tools');
+
+    const state = {
+        showOutline: false,
+        wireframe: false,
+        displayCharts: false,
+    };
+
+    // charts
+    gui.add(state, 'displayCharts').name('Display charts').onChange((newValue) => {
+        if (newValue) {
+            chartDiv.style.display = 'flex';
+        } else {
+            chartDiv.style.display = 'none';
+        }
+    });
+
+    function applyToNodeFirstMaterial(cb) {
+        scene.getMap().tiles.children[0].traverse((object) => {
+            if (object.materials) {
+                cb(object.materials[0]);
+            }
+        });
+        scene.renderScene3D();
+    }
+
+    // tiles outline
+    gui.add(state, 'showOutline').name('Show tiles outline').onChange((newValue) => {
+        scene.map.layersConfiguration.getGeometryLayers()[0].showOutline = newValue;
```

Comment:
Wireframe and outline/grid are properties of appearance that can be used as cartographic tools for users, notions of style / semantics and debug. This PR relates these properties in 2 commits. I don't find in the comments contrary arguments on the field of action of these properties. So there is an interest to factorize these functions in this PR or to splitter this PR and create a new one with the 2 commits concerned.

**Tools that seem to mainly concern the debug** :
* Charts with count nodes
* Console output when selecting tile (not unified with this PR and __DEBUG__ )
* Current camera helper and global view (not unified with this PR and __DEBUG__ )
* BBox helper (not unified with this PR and __DEBUG__ )
* Animation debug message (not unified with this PR and __DEBUG__ )

There's also other unanswered comment :
[uniform int debug](https://github.com/iTowns/itowns2/pull/238#issuecomment-280332466)